### PR TITLE
Add resend OTP feature for registration with OTP request limit

### PR DESCRIPTION
**Pull request by dheeraj-p**

### CODE REVIEW

File: user/registration/src/main/java/in/projecteka/jataayu/registration/ui/activity/RegistrationActivity.kt [hunk 1]
```diff
@@ -65,10 +71,18 @@ class RegistrationActivity : BaseActivity<RegistrationActivityBinding>() {
                     showErrorDialog(it.error.localizedMessage)
                 }
                 is PartialFailure -> {
-                    showAlertDialog(getString(R.string.failure), it.error?.message, getString(android.R.string.ok))
+                    val currentFragment = supportFragmentManager.findFragmentById(R.id.fragment_container);
```
Comment (AnusreeSajeevan) [inline user/registration/src/main/java/in/projecteka/jataayu/registration/ui/activity/RegistrationActivity.kt]: Check the error code for request exceeded and then based on that show the error

File: user/registration/src/main/java/in/projecteka/jataayu/registration/ui/fragment/RegistrationOtpFragment.kt [hunk 2]
```diff
@@ -40,30 +41,32 @@ class RegistrationOtpFragment : BaseFragment() {
...
-
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         binding.viewModel = viewModel
         parentVM.appBarTitle.set(getString(R.string.verification))
 
-        viewModel.otpMessageLbl.set(SpannableStringBuilder()
-            .append(getString(R.string.otp_sent))
-            .bold { append(" ${parentVM.getIdentifierValue()}") })
-
```
Comment (AnusreeSajeevan) [inline user/registration/src/main/java/in/projecteka/jataayu/registration/ui/fragment/RegistrationOtpFragment.kt:55]: Confirm why the identifier value has been removed.

File: user/registration/src/main/java/in/projecteka/jataayu/registration/ui/fragment/RegistrationOtpFragment.kt [hunk 3]
```diff
@@ -76,5 +79,14 @@ class RegistrationOtpFragment : BaseFragment() {
                 }
             }
         })
+
+        parentVM.requestVerificationResponseLiveData.observe(this, Observer {
+            when (it) {
+                is PartialFailure -> {
```
Comment (AnusreeSajeevan) [inline user/registration/src/main/java/in/projecteka/jataayu/registration/ui/fragment/RegistrationOtpFragment.kt:84]: check the error code and handle the error based on the error code

File: user/registration/src/main/res/layout/fragment_otp_verification.xml [hunk 4]
```diff
@@ -106,6 +106,19 @@
...
 
         </com.google.android.material.button.MaterialButton>
+
+        <TextView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            app:layout_constraintTop_toBottomOf="@+id/btn_continue"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            android:layout_marginTop="@dimen/content_margin_36"
```
Comment (mahabaleshwarhnr) [inline user/registration/src/main/res/layout/fragment_otp_verification.xml:116]: resend should be button right?

File: user/registration/src/main/java/in/projecteka/jataayu/registration/ui/fragment/RegistrationOtpFragment.kt (hunk 2 above)
Comment (dheeraj-p) [reply user/registration/src/main/java/in/projecteka/jataayu/registration/ui/fragment/RegistrationOtpFragment.kt:55]: I have added it in the next line. So there was already a TextView for identifier which was not being used. That's why we removed it from the `otpMessageLbl` and added it to `mobileNumberText`.

File: user/registration/src/main/java/in/projecteka/jataayu/registration/ui/activity/RegistrationActivity.kt (hunk 1 above)
Comment (dheeraj-p) [reply user/registration/src/main/java/in/projecteka/jataayu/registration/ui/activity/RegistrationActivity.kt]: Yes sure we will change that.

File: user/registration/src/main/res/layout/fragment_otp_verification.xml (hunk 4 above)
Comment (dheeraj-p) [reply user/registration/src/main/res/layout/fragment_otp_verification.xml:116]: Ideally yes. But in the mockups it looked like a textview

**Review by AnusreeSajeevan — APPROVED** [review]
Looks good.

**Pull request merged**